### PR TITLE
Linkedin Token API Security

### PR DESCRIPTION
**Pull request by aidanouckama**
A recent infiltration of our API resorted us to implement this new token system, which only provides users access to ONLY their profiles. 

Also fixed a few bugs along with this:
- `mongoDBUtil.js` did not have `fetchWithAuth` implemented.
- `Navbar.js` called the Linkedin API twice instead of a singular call.
- `fetchWithAuth` used the `sessionId` cookie instead of the `token` cookie

**Infiltration Case**:
A user used our `/api/token` endpoint to receive a token for Authorization to use the other API endpoints freely.

**Case Fix**:
Implemented a new element within the token payload `linkedinToken` which stores the Linkedin Token given to the user on login. Added a middleware that verifies this LinkedIn token against the requested profile being changed.

**Linkedin Token**
- `client/src/util/tokenUtils.js` - Added a `linkedinToken` parameter to be added to the payload of the user's token.
- `server/controllers/authController.js` - Updated our LinkedIn API endpoint to respond with a LinkedIn `access token` and `refresh token`. The `access token` is used to access Linkedin API calls and the `refresh token` is used to refresh the access token when the expiration date is reached. Also added a way to refresh the LinkedIn `access token` but no implementation is done yet.
- `server/middleware/token.js` - Added a `linkedinToken` to the payload of the token when generated.
- `server/routes/token.js` - Included the addition of a `linkedinToken` within the payload.
- [NEW] `server/middleware/user.js` - Middleware to verify the user matches the requested user to be updated.

**Backend Routes Structure**
- `server/routes/*` - Changed some routes to include two routers, a `write` router and a `read` router. `Write` routers need a separate router so they can have the `user.js` middleware applied to them. Should follow this structure if a controller has both read and write API calls. May change the structure in the future if a better solution is found.
- `server/index.js` - Implemented read and write routers to include their respective middleware.

**Clean Up**
- `server/controllers/authController.js` - Removed our old implementation of tokens (which wasn't even being verified...)
- `server/*` - Lint auto-changes (indentation / prettier fixes).

LONG LIVE PIPELINES 🚀

### CODE REVIEW

**Comment by aidanouckama** [comment]
> Your [Render](https://render.com) PR Server URL is https://pipelines-pr-139.onrender.com.
> 
> Follow its progress at https://dashboard.render.com/web/srv-co6puj7sc6pc73aqf36g.

Backend LGTM. Service is live!

File: client/src/components/Navbar.js [hunk 1]
```diff
@@ -111,50 +112,63 @@ const Navbar = () => {
...
+                    login(user.email)
+                }
+            } catch (error) {
+                console.error(error.message)
+            }
+        }
+
+        handleLinkedinToken()
+    }, [authCode]) // Run when authCode or user changes
+
```
Comment (Nyumat) [inline client/src/components/Navbar.js:161]: clean. 🔥

File: server/middleware/token.js [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
+
+  // If profileId exists, include it in the payload
+  if (linkedinToken) {
+    payload.linkedinToken = linkedinToken;
+  }
+
+  // Generate token with payload
+  const token = jwt.sign(payload, process.env.JWT_SECRET, {
+    expiresIn: "1d",
+  });
```
Comment (Nyumat) [inline server/middleware/token.js:15]: Yep. Fantastic!

**Review by Nyumat — APPROVED** [review]
to the moon baby. 🚀

File: client/src/components/Navbar.js (hunk 1 above)
Comment (aidanouckama) [reply client/src/components/Navbar.js:161]: This was the useEffect that had me up till 1:30 AM 😭

**Pull request merged**